app/assets/javascripts/config/angularRoutes.js.coffee

```diff
@@ -96,6 +100,8 @@
           templateUrl: 'account/templates/create-account.html'
           controller: 'AccountController'
       onEnter: ['AccountService', (AccountService) ->
+        AccountService.clearAccountMessages()
```

Comment:
@chrisdolendo yeah, unfortunately... we had already tried to refactor this into a stateChange callback if you remember (which seemed so smart), and I realized when doing this bugfix that our refactor had broken things. The order of operations is: `stateChangeStart`, then `resolve/onEnter`, then `stateChangeSuccess` ... the problem was that we can't `clearUserAuth` on `stateChangeSuccess` (otherwise it will wipe out other things that happen in `onEnter/resolve` functions in the routes) and we can't do it `stateChangeStart` otherwise it was too early -- can't remember the exact use cases but trust me I tried all different combinations for a while, and ended up putting things back into the routes rather than go down the 🐰  hole. 😞